src/gpt-auto-generator/gpt-auto-generator.c

```diff
@@ -447,6 +447,43 @@ static int add_esp(DissectedPartition *p) {
 }
 #endif
 
+static int add_root_rw(DissectedPartition *p) {
+        const char *path;
+        int r;
+
+        assert(p);
+
+        if (in_initrd()) {
+                log_debug("In initrd, not generating drop-in for systemd-remount-fs.service.");
+                return 0;
+        }
+
+        if (arg_root_rw >= 0) {
+                log_debug("Parameter ro/rw specified on kernel command line, not generating drop-in for systemd-remount-fs.service.");
+                return 0;
+        }
+
+        if (!p->rw) {
+                log_debug("Root partition marked read-only in GPT partition table, not generating drop-in for systemd-remount-fs.service.");
+                return 0;
+        }
+
+        path = strjoina(arg_dest, "/systemd-remount-fs.service.d/50-remount-rw.conf");
+        (void) mkdir_parents(path, 0755);
+
+        r = write_string_file(path,
```

Comment:
I don't like this at all. Drop-ins with environment variables are a terrible mechanism to convey information from one generator to another. Also the overriding of ConditionPathExists is smelly.

Let's consider an alternative approach: gpt-auto-generator *also* reads /etc/fstab, and simply scans it to look for "/". If it doesn't find it, it generates a proper `systemd-remount-fs.service` and pulls it into `local-fs.target`. The static enablement of `systemd-remount-fs.service` is disabled, and `systemd-fstab-generator` pulls it in when useful.